schemas.js

```diff
@@ -0,0 +1,25 @@
+/*
+ * Validation schemas for incoming requests
+*/
+
+var Joi = require('joi');
+var mimes = Object.keys(require('mime-db'));
+
+exports.file = Joi.object().keys({
+  type: Joi.string().valid(mimes),
```

Comment:
Don't we have restrictions on which files we'll accept into a project? This would be a good place to enforce that.